04-onion-routing.md

```diff
@@ -181,17 +181,14 @@ Each `hop_payload` has the following structure:
 
 1. type: `hop_payload`
 2. data:
-   * [`1`: `num_frames_and_realm`]
+   * [`1`: `realm`]
    * [`raw_payload_len`: `raw_payload`]
    * [`padding_len`: `padding`]
    * [`32`: `HMAC`]
 
 Notice that since the `hop_payload` is instantiated once per hop, the subscript `_i` may be used in the remainder of this document to refer to the `hop_payload` and its fields destined for hop `i`.
 
-The `hop_payload` consists of at least one `frame` followed by up to 15 additional `frame`s.
-The number of additional frames allocated to the current hop is determined by the 4 most significant bits of `num_frames_and_realm`, while the 4 least significant bits determine the payload format.
-Therefore the number of frames allocated to the current hop is given by `num_frames = (num_frames_and_realm >> 4) + 1`.
-For simplification we will use `hop_payload_len` to refer to `num_frames * FRAME_SIZE`.
+The `hop_payload` consists of at least one `frame` followed by up to 15 additional `frame`s.  For simplification we will use `hop_payload_len` to refer to `num_frames * FRAME_SIZE`.
```

Comment:
If the entire byte is available, why restrict it to 15 instead of ~20 which lets nodes utilize all the available space?